pootle/apps/pootle_app/management/commands/update_external_tmserver.py

```diff
@@ -0,0 +1,151 @@
+
+# This must be run before importing Django.
+os.environ['DJANGO_SETTINGS_MODULE'] = 'pootle.settings'
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+
+BULK_CHUNK_SIZE = 5000
+
+
+class Command(BaseCommand):
+    help = "Add external translations to Local Translation Memory"
+    option_list = BaseCommand.option_list + (
+        make_option('--rebuild',
+                    action="store_true",
+                    dest='rebuild',
+                    default=False,
+                    help='Drop the entire TM on start and update everything '
+                         'from scratch'),
+        make_option('--dry-run',
+                    action="store_true",
+                    dest='dry_run',
+                    default=False,
+                    help='Report the number of translations to index and quit'),
+        make_option('--tm',
+                    action='store',
+                    dest='tm_server',
+                    help="TM to use. TM must exist on settings. TM will be "
+                         "created on the server if it doesn't exist"),
+        make_option('--target-language',
+                    action='store',
+                    dest='target_language',
+                    help="Target language to fallback to use in case it can't "
+                         "be guessed for any of the input files."),
+    )
+
+    def _parse_translations(self, *args, **options):
+        units = []
+
+        for filename in args:
+            store = factory.getobject(filename)
+            if (not store.gettargetlanguage() and
+                not self.target_language and
+                not options['dry_run']):
+
+                raise CommandError("Unable to get target language for '%s'. "
+                                   "Please specify a fallback target language "
+                                   "with --target-language" % filename)
+            units.extend([unit for unit in store.units if unit.istranslated()])
+
+        total = len(units)
+
+        if total == 0:
+            self.stdout.write("No translations to index")
+            sys.exit()
+
+        self.stdout.write("%s translations to index" % total)
+
+        if options['dry_run']:
+            sys.exit()
+
+        self.stdout.write("")
+
+        for i, unit in enumerate(units, start=1):
+            if (i % 1000 == 0) or (i == total):
+                percent = "%.1f" % (i * 100.0 / total)
+                self.stdout.write("%s (%s%%)" % (i, percent), ending='\r')
+                self.stdout.flush()
+
+            target_language = unit.gettargetlanguage()
+            if target_language is None:
+                target_language = self.target_language
+
+            yield {
+                "_index": self.INDEX_NAME,
+                "_type": target_language,
+                "_id": unit.getid(),
+                'revision': 0,
+                'project': self.tm_settings['TM_NAME'],
```

Comment:
Will do since I brought back `--project` so people can push different files to a TM but specifying different projects. Thus a `libreoffice` TM can be pushed translations for project "LibreOffice 4.3 UI" and "LibreOffice 4.4 Docs". Or even mix completely different projects, for example Inkscape and XChat.